crates/supervisor/core/src/chain_processor/error.rs

```diff
@@ -0,0 +1,14 @@
+use crate::syncnode;
+use thiserror::Error;
+
+/// Errors that may occur while processing chains in the supervisor core.
+#[derive(Debug, Error)]
+pub enum ChainProcessorError {
+    /// Represents an error that occurred while interacting with the managed node.
+    #[error(transparent)]
+    ManagedNode(#[from] syncnode::ManagedNodeError),
```

Comment:
the name of this error is specific enough to import it in the module prelude already